src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -77,9 +84,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.basic());
     String recipient = request.getParameter("recipient");
 
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("image");
+
     Message message = new Message(user, text, recipient);
+
+    if (blobKeys != null && !blobKeys.isEmpty()) {
+      BlobKey blobKey = blobKeys.get(0);
+      ImagesService imagesService = ImagesServiceFactory.getImagesService();
+      ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
+      String imageUrl = imagesService.getServingUrl(options);
+      message.setImageUrl(imageUrl);
+    }
+
     datastore.storeMessage(message);
 
-    response.sendRedirect("/user-page.html?user=" + recipient);
```

Comment:
Did you intend to change this from recipient to user (Ching Ching changed it from user to recipient with one of her changes.)